### PR TITLE
Reset password users/recover

### PR DESCRIPTION
**Pull request by fileames**
Reset password endpoint is created. Closing #166 

When a user provides a email, if email exists in the database, an email with contents is sent:
```
Your new password for the account with username {user identifier} is {new password}, you can login using this password and change it on settings.
```

Subject of the email is `New Password for your Squad Game Account`

For security, user should be able to set the new password and the password should not be exposed, this improvement is opened as an issue and will be fixed later: #228

### CODE REVIEW

File: back-end/sports_platform/sports_platform_api/tests/test_recover.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+from django.test import Client, TestCase
+from ..models import User
+from rest_framework.authtoken.models import Token
+from django.contrib.auth import authenticate
+from .test_helper_functions import create_mock_user
+from django.core import mail
+
+class LoginTest(TestCase):
```
Comment (berkatil) [inline back-end/sports_platform/sports_platform_api/tests/test_recover.py:8]: I guess the class name should be changed

File: back-end/sports_platform/sports_platform_api/views/user_views.py [hunk 2]
```diff
@@ -90,3 +90,38 @@ def logout(request):
...
+        if res == 100:
+            # No user with email
+            return Response({"message": "If email provided is correct, a reset password is sent, please check spam."},
+                            status=200)
+        elif res == 500:
+            return Response({"message": "Try again."},
+                            status=500)
+
+    except Exception as e:
+        print(e)
```
Comment (berkatil) [inline back-end/sports_platform/sports_platform_api/views/user_views.py:122]: print should be deleted

**Review by berkatil — CHANGES_REQUESTED** [review]

File: back-end/sports_platform/sports_platform_api/tests/test_recover.py (hunk 1 above)
Comment (fileames) [reply back-end/sports_platform/sports_platform_api/tests/test_recover.py]: Changed, thanks for pointing out.

File: back-end/sports_platform/sports_platform_api/views/user_views.py (hunk 2 above)
Comment (fileames) [reply back-end/sports_platform/sports_platform_api/views/user_views.py]: Deleted

**Comment by berkatil** [comment]
Thanks for the efforrt

**Pull request merged**